model_analyzer/record/types/cpu_available_ram.py

```diff
@@ -57,6 +57,13 @@ def header(aggregation_tag=False):
 
         return ("Max " if aggregation_tag else "") + "RAM Available (MB)"
 
+    def calculate_percentage_gain(self, other: Record) -> float:
```

Comment:
It still feels like we should be able to define this in Record for the common case, and only overwrite it for cases where lower is better. We shouldn't have to shotgun 25 files. I understand that for add and sub they had to be manually specified because it had to return a new object of the specific type (which I still think could be solved in a better way), but this is just returning a float and is only utilizing value() function which is defined in Record

Reply:
It's a near 50/50 split (11 of 19 need to be inverted). Seems like it would be more work to rip it out at this point and I'm not sure what it buys us. I think it's more readable to have the function in the individual classes and we're less bug prone when adding future records.

Reply:
Any time we are cutting and pasting something over and over, we should find a better way. If we decide to change how this is calculated, now we have to update 25 places and are prone to error. Also, 25 instances of code technically needs to be tested 25 times.

To be clear, this was the basic design I was suggesting:

```
Record:
  _positive_is_better():
    undefined virtual function

  calculate_percentage():
    if _positive_is_better():
      <handle that case>
    else:
      <handle that case>

class IncreasingRecord(Record):
  _positive_is_better(): Return True

class DecreasingRecord(Record):
  _positive_is_better(): Return False

<All records extend either IncreasingRecord or DecreasingRecord. No need to cut and paste over and over>
```

Reply:
I see what you're saying. I've added this and unit testing to protect it. Please re-review